docs: PE-8757: Ubuntu 24.04 STIG - Edge (#10751)
spectrocloud/librarium: docs/docs-content/clusters/edge/edgeforge-workflow/palette-canvos/build-provider-images/build-provider-images.md

---
sidebar_label: "Build Provider Images"
title: "Build Provider Images"
description: "Learn how to build provider images using EdgeForge utilities."
icon: ""
hide_table_of_contents: false
sidebar_position: 0
tags: ["edge"]
---

Provider images are Kairos-based container images containing the OS and the desired Kubernetes version. These images
install an immutable OS and software dependencies compatible with a specific Kubernetes version during cluster
deployment. A provider image is used in the OS and the Kubernetes layer when creating a cluster profile. These container
images are downloaded during the installation by the Edge Installer and converted to disk images for the system to boot
into.

:::info

The provider images are one of the critical artifacts you need to build during EdgeForge. The other artifact is the Edge
Installer ISO. Both are required for Edge deployment. For education purposes, we provide separate instructions for
building the installer ISO and the provider images. However, these two artifacts are often built together in a single

> step in practice. Refer to [Build Edge Artifacts](../palette-canvos.md) for a how-to that covers how to build both

Reviewer: ```suggestion
step in practice. Refer to [Build Edge Artifacts](../palette-canvos.md) to learn how to build both
```

Reviewer: Maybe?

artifacts at the same time.

:::

## Specialized Build Guides

If you need security-hardened or platform-specific provider images, use one of the following guides instead of the
generic procedure on this page:

- [Build AWS Cloud Images](./build-aws-cloud-image.md)
- [Build MAAS Images](./build-maas-image.md)
- [Build RHEL 9 STIG Images](./build-rhel-stig-image.md)
- [Build Ubuntu 24.04 STIG Images](./build-ubuntu-stig-image.md)

## Build Custom Provider Images

The following procedure builds standard provider images using any supported OS and Kubernetes distribution. Use this

> guide when you do not require the specialized configurations above.

Reviewer: ```suggestion
guide when you do not require [specialized configurations](#specialized-build-guides).
```

Reviewer: Maybe? Trying to avoid directional guidance.
